packages/analytics-browser/src/plugins/destination.ts

```diff
@@ -0,0 +1,17 @@
+import { Destination as CoreDestination, buildResult } from '@amplitude/analytics-core';
+import { BrowserConfig, DestinationContext as Context } from '@amplitude/analytics-types';
+
+export class Destination extends CoreDestination {
+  config: BrowserConfig;
+
+  constructor(config: BrowserConfig) {
+    super();
+    this.config = config;
+  }
+
+  async fulfillRequest(list: Context[], code: number, message: string) {
+    await this.config.diagnosticProvider?.track(list.length, code, message);
+    this.saveEvents();
```

Comment:
This will save events to the `config.storageProvider` which could cause them to be picked up by the AmplitudeDestination plugin. I think we will need a separate storageProvider or maybe just a new storageKey.

Reply:
I feel it will not cause unwanted events to be saved. I only add one line to [this function](https://github.com/amplitude/Amplitude-TypeScript/blob/e5c1ebafc300e517e809f42b015a26adebe7d79b/packages/analytics-core/src/plugins/destination.ts#L289-L292). Originally these events should be saved.

As moving diagnosticProvider to core package, I add  `diagnosticProvider.track()` to core destination. Please see the latest change.